### ISSUE TITLE
Bad Default for OS X

### ISSUE THREAD

**Issue by runeimp** [simoncozens/sile#369]
My first try with SILE and I got the following:

```
$ sile hello
This is SILE 0.9.4
<hello.sil>

! Couldn't find face Gentium at hello
```

I guessed this was a font problem as I didn't find a single suggestion regarding this error via Google. Might want to use a default font actually preinstalled on OS X like _Geneva_. So I went in search of Gentium. I found _Gentium Plus_ and _Gentium Book_ fairly easy. Downloaded and installed those but no joy. Same error. So figured this might be a Freetype issue. Searched and searched and eventually found `fc-cache -r` for fontconfig. So refreshed the font cache and still no joy. So figured I'd search more and eventually found the [_Original_ Gentium](http://scripts.sil.org/cms/scripts/page.php?item_id=Gentium_download#e76f0759). This was not easy as it should have been. So I installed that and the Hello SILE example finally worked as expected.

```
$ sile hello
This is SILE 0.9.4
<hello.sil>
[1]
```

I look forward to playing more with SILE now that I have the missing font situation worked out. Please, update the default font to (at the minimum) _Gentium Plus_ if not a default found on the system in question. Homebrew only works on OS X. So shouldn't be too difficult to at least update the default there. May also want to update that error with some useful info like a URL to search for the missing font in question. Or some such. This typesetting app (which BTW, I love the idea behind) is going to get little traction if the first experience for most users is "WTF?".

**Comment by simoncozens** [comment]
Yes, I can see that's not a great first experience for people! Ideally the default should be a nice-looking font that's likely to be available on all platforms, but I don't think such a think exists.

We should at least move the default to Gentium Plus (and then update all the tests - since, joy of joys, Gentium Plus uses a different glyph id encoding to Gentium), and I like the idea of providing a more helpful error message if the default font can't be found. There should be something in the README and the docs about it too. Good catch!

**Comment by alerque** [comment]
One option would be to include _a font_ file with SILE as the fallback / default font. Another would be to fix the distribution packages to install the default font as a dependency. This is easy on Linux but can also be done in Homebrew I think.

**Comment by texrg** [comment]
How seting default font to other?
for example 'Lato' http://www.latofonts.com/lato-free-fonts/
in fedora not working too ;(

**Comment by alerque** [comment]
@texrg Fedora has packages for Gentium. In fact it  was included by default from versions 4-15, after which I  don't have data to go on. It seems to be still out there under thte package name `sil-gentium-fonts`. Do you have that installed?

Lato has nice glyph support, but my experience is the default versions that come with operating systems tend to be very different versions, so even though the availability is good the fact that the family went through a major overhaul as recently as a couple years ago make it a bad choice to assume everybody has the same thing.

I'm still of the mind we could bake one Serif font into SILE as the fallback. In the mean time you should _always_ specify what fonts _you_ want to use in your documents anyway. The fallback shouldn't really ever be hit out of the box.

**Comment by runeimp** [comment]
To be honest if "the fallback shouldn't really ever be hit out of the box" why would be care what the default font is? I think distributing the font with the package is reasonable. If "we" really care that much about the specific default then just include it with the app. Linux/Mac/Windows all support TTF fonts, right? So just include the one font with the package.

**Comment by simoncozens** [comment]
I'm not in favour of _bundling_ a default font with SILE; it makes the license situation that much more complicated. (Yeah, we do one already, and maybe that's a mistake.)

**Comment by texrg** [comment]
@alerque 
[fedora]

 ./sile ./minimal.sil 
This is SILE 0.9.5-unreleased
<./minimal.sil>
[1] 
! Font 'Gentium Plus' not available, falling back to 'FreeMono'

# dnf list --installed  *entium*
Zainstalowane pakiety [instaled package]
sil-gentium-alt-fonts.noarch                       1.02-18.fc23                         @fedora
sil-gentium-basic-book-fonts.noarch                1.1-13.fc23                          @fedora
sil-gentium-basic-fonts.noarch                     1.1-13.fc23                          @fedora
sil-gentium-basic-fonts-common.noarch              1.1-13.fc23                          @fedora
sil-gentium-fonts.noarch                           1.02-18.fc23                         @fedora
sil-gentium-fonts-common.noarch                    1.02-18.fc23                         @fedora

**Comment by qtfkwk** [comment]
tldr: `wget 'http://scripts.sil.org/cms/scripts/render_download.php?format=file&media_id=Gentium-1.03&filename=Gentium-1.03.zip' -O Gentium-1.0.3.zip; unzip Gentium-1.0.3.zip; find Gentium-1.03 -name \*.ttf |xargs open`

The statement "Default font: The default font used by SILE is Gentium Plus, available from here (http://software.sil.org/gentium/). If this font is not installed on your system, you won't be able to use the examples without modification." from https://github.com/simoncozens/sile/blob/master/README.md is simply not entirely true.

Possible fixes include updating it to say that the default font used by SILE is Original Gentium, available from here (http://scripts.sil.org/cms/scripts/page.php?item_id=Gentium_download). The direct link to the zip file you want is http://scripts.sil.org/cms/scripts/render_download.php?format=file&media_id=Gentium-1.03&filename=Gentium-1.03.zip."

Also, the issue probably deserves coverage in the manual.

**Comment by simoncozens** [comment]
I'm not sure why the statement is untrue. I think that as of 1f2a7b3, if SILE doesn't know what font to use it will use Gentium Plus. Are you talking about the latest release (in which case, yes, the github README doesn't apply - the Gentium Plus change hasn't landed in a release yet), or do you mean something else?

**Comment by qtfkwk** [comment]
I reread my comment and realize that it isn't very clear. So my issue was that the font name in the readme should be "Original Gentium", not "Gentium Plus". That issue is then exacerbated by vague links to a confusing site that doesn't really guide the user towards what they want, which is to install whichever font file(s) they need in order to make the SILE hello world work. This issue should be made clear by either SILE itself, by choosing a default or that is universal or built into SILE, or into its documentation, both the readme and the manual.

It's true that I don't understand issues regarding which versions/releases the readme applies to or not. My feedback is outside of those concerns, and is simply based on my initial experience trying to use SILE. I wanted to install it quickly on my Mac, preferably via Homebrew, and start using it immediately. As is, this is not possibly without immediately encountering problems with perhaps poorly-chosen defaults, and/or missing font files.

Please don't take my direct tone as attacking... I love the idea behind SILE! I only wish the entry was easier and more polished.

**Comment by simoncozens** [comment]
I don't feel attacked! I do want to make the user experience as smooth as possible, and I can see that not having the default font installed by default makes it hard to get started.

There seems to be a homebrew-fonts cask with a font-gentium-plus formula. I don't know if it's possible for a homebrew formula like SILE to depend on something in a cask, but the ideal case would be to automatically tap the cask and install the font. (Hopefully someone with more homebrew clue will be able to help me make that work if it is possible!) The second best case would be to run:

```
$ brew tap caskroom/fonts
$ brew cask install font-gentium-plus
```

This is on the basis that you are running `brew install sile --HEAD`, which is the version with Gentium Plus as the default.

**Comment by alerque** [comment]
I checked a few distros for Gentium Plus packages to see if requiring it as a package dependency is reasonable.

* Homebrew has [a font cask](https://github.com/Homebrew/homebrew-cask-fonts/blob/master/Casks/font-gentium-plus.rb) 
* Ubuntu at least as far back as previous now obsolete 16.04 LTS release also has [packages](https://packages.ubuntu.com/xenial/fonts-sil-gentiumplus) for Gentium Plus
* Arch has [packages](https://aur.archlinux.org/packages/ttf-gentium-plus/) for Gentium Plus
* Debian has [packagen](https://packages.debian.org/sid/fonts/fonts-sil-gentiumplusg

According to [Repology](https://repology.org/project/fonts:gentium-plus/versions) there are a few more.

Notably Fedora is a little concerning as it seems to have Gentium Basic packages as if it was Plus. Maybe Plus is bundled in there too?. @texrg do you have any insight as to what's going on there? Well there be a problem packaging SILE for Fedora if we expect that font as a system package dependency?

@runeimp Even though I suggested it, bundling a font with the app is actually a little bit problematic because we would not be able to place the font on the system along with all other fonts (this would cause conflicts on many systems), so we would have to keep it in some private location with SILE's code. Then we'd have to introduce a fallback mechanism that could resort to finding and loading fonts from our private stash instead of the usual system paths. This introduces complexity and another mechanism that could fail or give unexpected results.
